public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/azure_monitor_datasource.ts

```diff
@@ -105,11 +105,11 @@ export default class AzureMonitorDatasource extends DataSourceWithBackend<AzureM
     const dimensionFilters = (item.dimensionFilters ?? [])
       .filter((f) => f.dimension && f.dimension !== 'None')
       .map((f) => {
-        const filter = templateSrv.replace(f.filter ?? '', scopedVars);
+        const filter = f.filter?.map((filter) => templateSrv.replace(filter ?? '', scopedVars)).filter((item) => item!);
```

Comment:
Can this last filter be moved to the already existing filter on line 106 so that we don't iterate over all items another time?